src/Endpoints/TenantToken.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MeiliSearch\Endpoints;
+
+use DateTime;
+use MeiliSearch\Contracts\Endpoint;
+use MeiliSearch\Exceptions\InvalidArgumentException;
+use MeiliSearch\Http\Serialize\Json;
+
+class TenantToken extends Endpoint
+{
+    private function base64url_encode($data)
+    {
+        return rtrim(strtr(base64_encode($data), '+/', '-_'), '=');
+    }
+
+    /**
+     * Generate a new tenant token.
+     *
+     * The $options parameter is an array, and the following keys are accepted:
+     * - apiKey: The API key parent of the token. If you leave it empty the client API Key will be used.
+     * - expiresAt: A DateTime when the key will expire. Note that if an expiresAt value is included it should be in UTC time.
+     */
+    public function generateTenantToken($searchRules, ?array $options = []): string
```

Comment:
Maybe the function could be shorten by putting the validation in another one. It is a bit lengthy but it may also be normal in PHP in which case its good for me :D